src/lm_robust_helper.cpp

```diff
@@ -14,44 +14,46 @@ Eigen::MatrixXd AtA(const Eigen::MatrixXd& A) {
 }
 
 // [[Rcpp::export]]
-List lm_solver(Eigen::Map<Eigen::MatrixXd>& Xfull,
-               const Eigen::Map<Eigen::VectorXd>& y,
-               const Rcpp::Nullable<Rcpp::NumericMatrix> & Xunweighted,
-               const Rcpp::Nullable<Rcpp::NumericVector> & weight,
-               const double & weight_mean,
-               const Rcpp::Nullable<Rcpp::IntegerVector> & cluster,
-               const int & J,
-               const bool & ci,
-               const String type,
-               const std::vector<bool> & which_covs,
+Eigen::MatrixXd Kr(const Eigen::MatrixXd& A, const Eigen::MatrixXd& B) {
+  Eigen::MatrixXd AB(A.rows() * B.rows(), A.cols() * B.cols());
+
+  for (int i = 0; i < A.rows(); i++) {
+    for (int j = 0; j < A.cols(); j++) {
+      AB.block(i*B.rows(), j*B.cols(), B.rows(), B.cols()) = A(i, j) * B;
+    }
+  }
+  return AB;
```

Comment:
somewhat surprised you had to roll your own kronecker product.

Reply:
Unsupported 🙌 https://eigen.tuxfamily.org/dox/unsupported/group__KroneckerProduct__Module.html